lib/spack/spack/compilers/fj.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import re
+from spack.compiler import Compiler
+
+class Fj(Compiler):
+    # Subclasses use possible names of C compiler
+    cc_names = ['fcc']
+
+    # Subclasses use possible names of C++ compiler
+    cxx_names = ['FCC']
+
+    # Subclasses use possible names of Fortran 77 compiler
+    f77_names = ['frt']
+
+    # Subclasses use possible names of Fortran 90 compiler
+    fc_names = ['frt']
+
+    # Named wrapper links within build_env_path
+    link_paths = {'cc': 'fj/fcc',
+                  'cxx': 'fj/FCC',
+                  'f77': 'fj/frt',
+                  'fc': 'fj/frt'}
+
+    @property
+    def openmp_flag(self):
+        return "-Kopenmp"
+
+    @property
+    def cxx98_flag(self): 
+        return "-std=c++98"
+
+    @property
+    def cxx11_flag(self):
+        return "-std=c++11"
+
+    @property
+    def cxx14_flag(self):
+        return "-std=c++14"
+
+    @property
+    def pic_flag(self):
+        return "-fPIC"
+
+    @classmethod
+    def default_version(cls, cc):
+        output = spack.compiler.get_compiler_version_output(cc, '--version')
+        match = re.search(r'(?:fcc \(FCC\) )?([\d.]+)', output)
+        version = match.group(match.lastindex) if match else 'unknown'
+        return version
+
+    @classmethod
+    def fc_version(cls, fc):
+        output = spack.compiler.get_compiler_version_output(fc, '-V')
+        match = re.search(r'(?:frt\: Fujitsu Fortran Driver Version )?([\d.]+)', output)
+        version = match.group(match.lastindex) if match else 'unknown'
+        return version
```

Comment:
I believe @alalazo just updated the compiler logic so that these methods are no longer necessary. You just need a single regex string to get the compiler version. See #10931 for examples. You can also add unit tests now! 😃

Reply:
The reason for creating methods is because there are differences about output of versions between Fujitsu C compiler and Fujitsu fortran compiler.

・Fujitsu C compiler
　$ fcc --version
　fcc (FCC) 1.2.0 20180907
　simulating gcc version 4.1.2
　Copyright FUJITSU LIMITED 2012-2018

・Fujitsu fortran compiler
　$ frt -V
　frt: Fujitsu Fortran Driver Version 1.2.0 (Sep  7 2018 18:26:33)

I'll come back after checking if compiler developers modify about output of versions.

Reply:
@alalazo is there a good way to handle this ^? I know there are other compilers in Spack with a similar problem. Either way, we should prob add unit tests now so we can test that the regex is working properly.

Reply:
Thank you for your advice.
I added unit tests for Fujitsu compiler on commit of "Add compiler tests for Fujitsu compiler", 
then all checks have passed! (Travis CI, codecov)

And, I asked the compiler developer, it was said that the compiler version output commands and formats will be unified between Fujitsu C compiler and Fujitsu fortran compiler in the future.